File: src/frontend/lib/application/dto/stop.dart
class StopNameDTO {
    final String stopName;

    StopNameDTO({
        required this.stopName,
    });

    factory StopNameDTO.fromJson(Map<String, dynamic> json) => StopNameDTO(
        stopName: json['name'],
    );
}

class TransitPathDTO {
    final String city;
    final String routeId;
    final int directionId;
    final String stopIdOrigin;
    final String stopIdDestination;

    TransitPathDTO({
        required this.city,
        required this.routeId,
        required this.directionId,
        required this.stopIdOrigin,
        required this.stopIdDestination,
    });

    factory TransitPathDTO.fromJson(Map<String, dynamic> json) => TransitPathDTO(
        city: json['city'],
        routeId: json['route_id'],
        directionId: json['direction_id'],
        stopIdOrigin: json['stop_id__origin'],
        stopIdDestination: json['stop_id__destination'],
    );


    Map<String, dynamic> toJson() => {
        'city': city,
        'route_id': routeId,
        'direction_id': directionId,
        'stop_id__origin': stopIdOrigin,
        'stop_id__destination': stopIdDestination,
    };
}

File: backend/application/dto/stop.py
from pydantic import BaseModel, Field

from backend.application.dto.route import ConveyanceDTO
from backend.domain.gtfs_rt.enums import City


class StopNameDTO(BaseModel):
    name: str = Field(description="Stop Name")


class NearbyQueryDTO(BaseModel):
    latitude: float = Field(description="Latitude", ge=-90.0, le=90.0)
    longitude: float = Field(description="Longitude", ge=-180.0, le=180.0)
    radius_m: int = Field(
        default=800, description="Search radius in meters", gt=0, le=5000
    )
    limit: int = Field(default=10, description="Maximum stops returned", gt=0, le=50)


class NearbyStopDTO(BaseModel):
    name: str = Field(description="Stop Name")
    distance_m: int = Field(description="Distance from the query point in meters")
    latitude: float = Field(description="Latitude of the nearest matching platform")
    longitude: float = Field(description="Longitude of the nearest matching platform")
    routes: list[ConveyanceDTO] = Field(description="Routes serving the stop")


class TransitPathDTO(BaseModel):
    city: City = Field(description="City")
    route_id: str = Field(description="Route ID")
    direction_id: int = Field(description="Direction ID")
    stop_id__origin: str = Field(description="Stop ID Origin")
    stop_id__destination: str = Field(description="Stop ID Destination")
